swarm/network/stream/syncer.go

```diff
@@ -111,13 +111,13 @@ func (s *SwarmSyncerServer) SetNextBatch(from, to uint64) ([]byte, uint64, uint6
 		timerC       <-chan time.Time
 	)
 
-	defer func() {
-		metrics.GetOrRegisterResettingTimer("syncer.set-next-batch.total-time", nil).UpdateSince(batchStart)
-		metrics.GetOrRegisterCounter("syncer.set-next-batch.batch-size", nil).Inc(int64(batchSize))
+	defer func(start time.Time) {
+		metrics.GetOrRegisterResettingTimer("syncer.set-next-batch.total-time", nil).UpdateSince(start)
+		metrics.GetOrRegisterGauge("syncer.set-next-batch.batch-size", nil).Update(int64(batchSize))
```

Comment:
Not sure about that - you have many syncers in a given process, so having a counter gives you the opportunity to count all batch sizes and sum them up.

Having this as a Gauge, which all syncers will modify doesn't make much sense to me. A Gauge is a singleton so to say.

Reply:
But the counter doesn't allow me to see the individual increments, i.e. I can't see the individual batch size

Reply:
A Gauge won't allow you to see the individual changes either.

Reply:
AFAIK Gauge will, but if I `sum` the gauge readings I'll get to the same measurement as the `Counter`s

Reply:
Gauge is losing the updates you make during the resolution time (which is 10sec.)

Reply:
done :+1: